gpflow/utilities/utilities.py

```diff
@@ -30,6 +30,7 @@
 Path = str
 Accumulator = Tuple[Path, State]
 TraverseUpdateCallable = Callable[[TraverseInput, Path, State], State]
+M = TypeVar('M', bound=tf.Module)
```

Comment:
What's the advantage of using this as a type annotation instead of `tf.Module` directly? If it *is* necessary, how about a more descriptive name (e.g. `ModuleType`)?

Reply:
type variables are more specific https://docs.python.org/3/library/typing.html#typing.TypeVar

they're typically single (or few) capital letters. it's not ideal in python since they're global, but that certainly the standard in other languages, and looks so for the docs too. I'm not sure of the benefit of moving it away from the function it's used in, but I don't feel strongly about that

Reply:
@st--, that change wasn't introduced by me. It was already in the code.

Reply:
I think it'd be much more helpful to have it right next to the function that it is used in (`deepcopy[_components]`), as in that context it's easily understood what it does, whereas having it at the top of the module makes both the definition and its usage more confusing.

Reply:
@awav I'd be happy to keep this in but I think it should go back next to the function it's used for.